extensions/amp-story-shopping/0.1/amp-story-shopping.js

```diff
@@ -3,6 +3,36 @@ import {AmpStoryShoppingConfig} from './amp-story-shopping-config';
 import {AmpStoryShoppingTag} from './amp-story-shopping-tag';
 
 import {CSS as shoppingCSS} from '../../../build/amp-story-shopping-0.1.css';
+import {dependsOnStoryServices} from '../../amp-story/1.0/utils';
+
+/**
+ * Language and currency sensitive number formatting.
+ * @param {!../../../src/service/localization.LocalizationService} localizationService
+ * @param {!Element} el
+ * @param {string} currency An ISO 4217 currency code.
+ * @param {number} price
+ * @return {string}
+ */
+export const formatI18nNumber = (localizationService, el, currency, price) => {
+  const langCode = localizationService.getLanguageCodesForElement(el)[0];
+  return new Intl.NumberFormat(langCode, {style: 'currency', currency}).format(
```

Comment:
What's displayed if the parameters are incorrect, e.g. for `currency === 'ZZZ'`?
Can you write unit tests?

Reply:
We need to make sure it doens't stop the execution of the code and still renders something (or nothing at all), but we should control what happens. Do we need a try/catch block?
I'm not sure how you'd validate if the currency is valid, e.g. ZZZ

Reply:
Added a fallback to render the number. 
To validate we will need to check against a [list of ISO 4217 Codes](https://github.com/merlos/iso4217-validator/blob/master/lib/iso4217/validator/valid_codes.rb).